### PR TITLE
Support otel tracing in cri remote image service

### PR DESCRIPTION
**Pull request by swagatbora90**
Signed-off-by: Swagat Bora <sbora@amazon.com>

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Extend support for tracing in CRI image service
#### Which issue(s) this PR fixes:
https://github.com/kubernetes/kubernetes/issues/112708

Fixes #

#### Does this PR introduce a user-facing change?
NONE

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
[KEP](https://github.com/kubernetes/enhancements/pull/2832)

### CODE REVIEW

**Comment by dims** [comment]
@swagatbora90 looks like you are missing the CLA as well

**Comment by swagatbora90** [comment]
> @swagatbora90 looks like you are missing the CLA as well

Yeah, i am working on it now.

File: pkg/kubelet/cri/remote/remote_image_test.go
Comment (SergeyKanzhelev) [inline pkg/kubelet/cri/remote/remote_image_test.go:2]: nit: fix the year

Comment (swagatbora90) [reply pkg/kubelet/cri/remote/remote_image_test.go]: fixed

**Comment by dims** [comment]
cc @endocrimes

File: pkg/kubelet/cri/remote/remote_image_test.go
Comment (dims) [reply pkg/kubelet/cri/remote/remote_image_test.go]: @SergeyKanzhelev other than this, does it look good?

**Review by endocrimes — APPROVED** [review]
/lgtm

thank you!

File: pkg/kubelet/cri/remote/remote_image_test.go
Comment (SergeyKanzhelev) [reply pkg/kubelet/cri/remote/remote_image_test.go]: sorry dropped the ball on this =). Looking

**Comment by dims** [comment]
thanks @endocrimes @SergeyKanzhelev 

/approve
/hold 

@swagatbora90 please remove hold when we have an LGTM from Sergey.

File: pkg/kubelet/cri/remote/remote_image.go
Comment (SergeyKanzhelev) [inline pkg/kubelet/cri/remote/remote_image.go:70]: I looked at couple places - it looks like context used for these calls is often a new context. Is the plan to change all the callsite to start using an incoming context?

Comment (SergeyKanzhelev) [reply pkg/kubelet/cri/remote/remote_image.go:70]: I'm not against this PR, I think it is a good step forward. I just want to make sure there are follow up work planned to improve traces correlation

Comment (swagatbora90) [reply pkg/kubelet/cri/remote/remote_image.go:70]: Yes, you are right. This  would only enable creation of individual traces for every request made to the runtime. For proper tracing support, we would need to instrument spans that could cover multiple gRPC calls. For example, instrumenting containerRuntime.SyncPod or even kubelet syncPod would cover multiple CRI calls which would produce more valuable traces. 

I am interested to work on the follow up tasks. The initial KEP#2832  mainly targeted tracing around http and grpc calls. And I don't see any existing issue or discussion around extending opentelemetry tracing support in kubelet for trace correlation. What will be the best way to start this discussion? Should I create an issue identifying the gaps in tracing in kubelet?

Comment (SergeyKanzhelev) [reply pkg/kubelet/cri/remote/remote_image.go:70]: issue is a good place to discuss it.

I understand the minimal step taken here

/lgtm
/unhold

**Pull request merged**
